### PR TITLE
Allow method parameter override for training based indices (solves issue #2246)

### LINKED ISSUE #2246: Allow method parameter override for training based indices
## Description

From mode/disk-based feature, we introduced concept of mode and compression level that will dynamically set default resolution. For indices created directly from the method (i.e. mapping) we allow users to override parameters if they want.

However, for training, we added the mode and compression, but made them mutually exclusive with method. See: https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/plugin/rest/RestTrainModelHandler.java#L138-L139. We need to remove this and allow method overriding to provide a consistent ux experience across the plugin.

### PR DESCRIPTION
**Pull request by buddharajusahil**
### Description
https://github.com/opensearch-project/k-NN/issues/2246

From mode/disk-based feature, we introduced concept of mode and compression level that will dynamically set default resolution. For indices created directly from the method (i.e. mapping) we allow users to override parameters if they want.

However, for training, we added the mode and compression, but made them mutually exclusive with method. See: https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/plugin/rest/RestTrainModelHandler.java#L138-L139. We need to remove this and allow method overriding to provide a consistent ux experience across the plugin.

### Related Issues
Resolves #2246 

### Check List
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [N/A] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/knn/plugin/action/RestTrainModelHandlerIT.java [hunk 1]
```diff
@@ -22,15 +23,7 @@
...
-import static org.opensearch.knn.common.KNNConstants.ENCODER_PARAMETER_PQ_CODE_SIZE;
-import static org.opensearch.knn.common.KNNConstants.ENCODER_PARAMETER_PQ_M;
-import static org.opensearch.knn.common.KNNConstants.KNN_ENGINE;
-import static org.opensearch.knn.common.KNNConstants.METHOD_ENCODER_PARAMETER;
-import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER_NLIST;
-import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER_SPACE_TYPE;
-import static org.opensearch.knn.common.KNNConstants.MODEL_ID;
-import static org.opensearch.knn.common.KNNConstants.NAME;
-import static org.opensearch.knn.common.KNNConstants.PARAMETERS;
+import static org.opensearch.knn.common.KNNConstants.*;
```
Comment (owenhalpert) [inline src/test/java/org/opensearch/knn/plugin/action/RestTrainModelHandlerIT.java]: Same as above

**Comment by navneet1v** [comment]
@owenhalpert can you please fix the changelog.

**Comment by heemin32** [comment]
Do we know why this was previously blocked, considering it only involves removing two lines of code?

**Comment by buddharajusahil** [comment]
> Do we know why this was previously blocked, considering it only involves removing two lines of code?

Hi Heemin,

So Jack said that these checks were added in this pr https://github.com/opensearch-project/k-NN/pull/2034 and not removed before the release. Then it got removed in this pr https://github.com/opensearch-project/k-NN/pull/2083 but training was missed

**Comment by jmazanec15** [comment]
@buddharajusahil Can you run `./gradlew :spotlessApply` to fix style?

**Review by jmazanec15 — APPROVED** [review]
LGTM - we just need to fix CI

**Review by jmazanec15 — APPROVED** [review]
LGTM - thanks

**Review by kotwanikunal — APPROVED** [review]

**Review by naveentatikonda — APPROVED** [review]

**Comment by naveentatikonda** [comment]
@buddharajusahil can you fix the conflicts and create a backport PR to 2.x ?

**Pull request merged**